server/plugin_release.go

```diff
@@ -29,199 +35,406 @@ import (
+	}
 
-	filepath, err := downloadAsset(assetURL, tempFolder)
-	if err != nil {
-		return err
+	// Verify signatures.
+	if err := verifySignatures(filePaths); err != nil {
+		return fmt.Errorf("failed signature verification err=%w", err)
 	}
 
-	fileServerPath, err := copyFileToSigningServer(filepath, tempFolder)
-	if err != nil {
-		return err
+	// All is well, remove *.tar.gz files from remote server.
+	if err := removeFilesFromRemoteServer(cfg, remotePaths); err != nil {
+		return fmt.Errorf("failed to remove files from remote server err=%w", err)
 	}
 
-	err = signAsset(fileServerPath)
+	return nil
+}
+
+// copyFilesFromRemoteServer copies remoteFiles to pluginFolder.
+func copyFilesFromRemoteServer(cfg *MatterbuildConfig, remoteFiles []string, pluginFolder string) error {
+	LogInfo("Copying files from remote server")
+
+	sftp, err := getPluginSigningSftpClient(cfg)
 	if err != nil {
 		return err
 	}
+	defer sftp.Close()
 
-	filename := strings.Split(filepath, "/")
-	baseFilename := filename[len(filename)-1]
-	localSignedFiles, err := copySignedFile(baseFilename, tempFolder)
-	if err != nil {
-		return err
+	for _, remoteFile := range remoteFiles {
+		srcFile, err := sftp.Open(remoteFile)
+		if err != nil {
+			return err
+		}
+		defer srcFile.Close()
+
+		destination := filepath.Join(pluginFolder, filepath.Base(remoteFile))
+		LogInfo("%s -> %s", remoteFile, destination)
+		dstFile, err := os.Create(destination)
+		if err != nil {
+			return err
+		}
+		defer dstFile.Close()
+
+		if _, err := srcFile.WriteTo(dstFile); err != nil {
+			return fmt.Errorf("error while reading from remote buffer err=%w", err)
+		}
 	}
 
-	err = uploadSignedArtifcatsToGithub(ctx, client, repo, releaseID, tempFolder, localSignedFiles)
+	LogInfo("Done copying files from remote server")
+	return nil
+}
+
+// copyFilesToRemoteServer copies files to the remote server.
+func copyFilesToRemoteServer(cfg *MatterbuildConfig, filePaths []string) ([]string, error) {
+	LogInfo("Copying files to the signing server")
+	var result []string
+
+	sftp, err := getPluginSigningSftpClient(cfg)
 	if err != nil {
-		return err
+		return nil, err
+	}
+	defer sftp.Close()
+
+	for _, filePath := range filePaths {
+		f, err := os.Open(filePath)
+		if err != nil {
+			return nil, err
+		}
+		defer f.Close()
+
+		serverPath := filepath.Join("/tmp", filepath.Base(filePath))
+		LogInfo("%s -> %s", filePath, serverPath)
+
+		// Open the source file
+		srcFile, err := sftp.Create(serverPath)
+		if err != nil {
+			return nil, err
+		}
+		defer srcFile.Close()
+
+		if _, err := srcFile.ReadFrom(f); err != nil {
+			return nil, err
+		}
+
+		result = append(result, serverPath)
 	}
 
-	err = uploadSignedArtifcatsToS3(localSignedFiles, tempFolder)
+	LogInfo("Done copying")
+	return result, nil
+}
+
+// removeFilesFromRemoteServer removes files from the remote server..
+func removeFilesFromRemoteServer(cfg *MatterbuildConfig, remoteFiles []string) error {
```

Comment:
I really like the self-explanatory function names, maybe we can remove the comment?